app/src/main/java/io/github/androidacademyglobal/ui/courses/CoursesViewModel.kt

```diff
@@ -0,0 +1,22 @@
+package io.github.androidacademyglobal.ui.courses
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import io.github.aag.core.domain.OperationResult
+import io.github.aag.core.domain.models.Course
+import io.github.aag.core.domain.repositories.CourseRepository
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+class CoursesViewModel(
+    private val courseRepository: CourseRepository
+) : ViewModel() {
+    init {
+        viewModelScope.launch {
+            courseRepository.loadAllCourses()
```

Comment:
Do you want to load courses at every viewmodel creation? I think it should be requested less frequent

Reply:
You mean "load via network" less frequently? I thought that repository would get data from DB while WorkManager task would update the cache periodically (once a day maybe) 🤔

Reply:
At this moment `loadAllCourses` loads only from remote data source

Reply:
Yeah, that's true. DB/caching/paging will be done within a separate step.